### PR TITLE
refactor(query): no need to deserialize all userinfo when create user

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

1. get_users need to record errlog when can not deserialize but do not retrun err

2. In create user, check user quota should use get_user_nums, ignore the deserialize.

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [x] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [x] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15671)

### CODE REVIEW

File: src/query/users/src/user_mgr.rs [hunk 1]
```diff
@@ -133,6 +133,16 @@ impl UserApiProvider {
         }
     }
 
+    // Get the tenant all user nums.
+    #[async_backtrace::framed]
+    pub async fn get_user_nums(&self, tenant: &Tenant) -> Result<usize> {
```
Comment (flaneur2020) [inline src/query/users/src/user_mgr.rs:138]: get_user_nums -> count_users? quq

File: src/query/management/src/user/user_api.rs [hunk 2]
```diff
@@ -27,6 +28,8 @@ pub trait UserApi: Sync + Send {
 
     async fn get_users(&self) -> Result<Vec<SeqV<UserInfo>>>;
 
+    async fn get_raw_users(&self) -> Result<ListKVReply>;
```
Comment (flaneur2020) [inline src/query/management/src/user/user_api.rs:31]: add a comment about why this method is introduced?

Comment (TCeason) [reply src/query/management/src/user/user_api.rs:32]: Ok

File: src/query/users/src/user_mgr.rs (hunk 1 above)
Comment (TCeason) [reply src/query/users/src/user_mgr.rs]: Delete this function. Directly call user_api.get_raw_users

**Review by drmingdrmer — COMMENTED** [review]
> Reviewed 2 of 7 files at r1, 1 of 6 files at r2, 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15671)* status: all files reviewed, 5 unresolved discussions (waiting on @flaneur2020 and @TCeason)

___
*[`src/query/management/src/role/role_mgr.rs` line 201 at r1](https://reviewable.io/reviews/datafuselabs/databend/15671#-Nz1IBcOCpOVQ5C1SMkN:-Nz1IBcOCpOVQ5C1SMkO:b9pu544) ([raw file](https://github.com/datafuselabs/databend/blob/f400cdb6518fd0d77c39d8ef9cacbf793a7a9250/src/query/management/src/role/role_mgr.rs#L201)):*
> ```XML
>     #[async_backtrace::framed]
>     #[minitrace::trace]
>     async fn get_role_nums(&self) -> Result<usize, ErrorCode> {
> ```

Use `RoleMgr::get_roles().len()` could be quite enough?
___
*[`src/query/management/src/role/role_mgr.rs` line 201 at r3](https://reviewable.io/reviews/datafuselabs/databend/15671#-Nz1sbgd5EjwpAviIvEU:-Nz1sbgd5EjwpAviIvEV:b9vrjsl) ([raw file](https://github.com/datafuselabs/databend/blob/883aba785ffc719a7af9ea91ad7e584a0cfc9eb9/src/query/management/src/role/role_mgr.rs#L201)):*
> ```XML
>     #[async_backtrace::framed]
>     #[minitrace::trace]
>     async fn get_raw_roles(&self) -> Result<ListKVReply, ErrorCode> {
> ```

Since there are builtin roles, this method should also be renamed to `get_raw_meta_roles`
___
*[`src/query/management/src/user/user_mgr.rs` line 163 at r1](https://reviewable.io/reviews/datafuselabs/databend/15671#-Nz1IRrqBVJhhQsvY5O8:-Nz1IRrqBVJhhQsvY5O9:b-72zukv) ([raw file](https://github.com/datafuselabs/databend/blob/f400cdb6518fd0d77c39d8ef9cacbf793a7a9250/src/query/management/src/user/user_mgr.rs#L163)):*
> ```XML
>     #[async_backtrace::framed]
>     #[minitrace::trace]
>     async fn get_user_nums(&self) -> Result<usize> {
> ```

Replace with `UserMgr::get_users().len()`?

**Comment by TCeason** [comment]
> ```
> get_user_nums
> ```

get_user_nums and get_role_nums already drop in the newest commit.

**Comment by TCeason** [comment]
> Since there are builtin roles, this method should also be renamed to get_raw_meta_roles

Good idea. I will rename it.

**Review by drmingdrmer — APPROVED** [review]
> Reviewed 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15671)* status: all files reviewed, 3 unresolved discussions (waiting on @flaneur2020 and @TCeason)

**Comment by TCeason** [comment]
cc @BohuTANG Now all ci passed. Please merge this pr.

**Pull request merged**